tyro/_cli.py

```diff
@@ -345,4 +401,10 @@ def fix_arg(arg: str) -> str:
 
     if dummy_wrapped:
         out = getattr(out, _strings.dummy_field_name)
-    return out
+
+    if return_unknown_args:
+        assert unknown_args is not None, "Should have parsed with `parse_known_args()`"
+        return out, unknown_args
+    else:
+        assert unknown_args is None, "Should have parsed with `parse_args()`"
+        return out
```

Comment:
perhaps we could just always return `out, unknown_args` here? since there's already an if statement in `cli()` the logic for ignoring `unknown_args` could be just placed up there

Reply:
We can return `out, unknown_args` here but we'd still need to keep some argument to know whether to call `parse_args` or `parse_known_args`. It would then seem a little odd to call this argument `return_unknown_args`. We could rename it to something else or just keep it as is?

Reply:
Aha, that's fair. The extra conditions hurt but so would an inconsistent argument name; I'm down for keeping what you have.